demos/coreMQTT/mqtt_demo_keep_alive.c

```diff
@@ -0,0 +1,1177 @@
+                        xDemoStatus = pdFAIL;
+                    }
+                }
+
+                /* Leave Connection Idle for some time. */
+                LogInfo( ( "Keeping Connection Idle..." ) );
+                vTaskDelay( mqttexampleDELAY_BETWEEN_PUBLISHES );
+            }
+        }
+
+        /********************** Unsubscribe from the topic. ***********************/
+
+        if( xDemoStatus == pdPASS )
+        {
+            LogInfo( ( "Unsubscribe from the MQTT topic %s.", mqttexampleTOPIC ) );
+            xDemoStatus = prvMQTTUnsubscribeFromTopic( &xMQTTContext );
+        }
+
+        /* Process Incoming packet from the broker. */
+        if( xDemoStatus == pdPASS )
+        {
+            xMQTTStatus = MQTT_ReceiveLoop( &xMQTTContext, mqttexampleRECEIVE_LOOP_TIMEOUT_MS );
+
+            if( xMQTTStatus != MQTTSuccess )
+            {
+                LogError( ( "MQTT_ReceiveLoop() failed with status %s.",
+                            MQTT_Status_strerror( xMQTTStatus ) ) );
+                xDemoStatus = pdFAIL;
+            }
+        }
+
+        /**************************** Disconnect. *****************************/
+
+        /* Whether the demo operations preceeding are successful or not, we
+         * always disconnect an open connection to free up system resources. */
+        if( xIsConnectionEstablished == pdTRUE )
+        {
+            /* Send an MQTT Disconnect packet over the already connected TCP socket.
+             * There is no corresponding response for the disconnect packet. After
+             * sending the disconnect, the client must close the network connection. */
+            LogInfo( ( "Disconnecting the MQTT connection with %s.",
+                       democonfigMQTT_BROKER_ENDPOINT ) );
+            xMQTTStatus = MQTT_Disconnect( &xMQTTContext );
+
+            if( xMQTTStatus != MQTTSuccess )
+            {
+                LogError( ( "MQTT_Disconnect() failed with status %s.",
+                            MQTT_Status_strerror( xMQTTStatus ) ) );
+                xDemoStatus = pdFAIL;
+            }
+
+            /* Close the network connection.  */
+            xNetworkStatus = SecureSocketsTransport_Disconnect( &xNetworkContext );
+
+            if( xNetworkStatus != TRANSPORT_SOCKET_STATUS_SUCCESS )
+            {
+                LogError( ( "SecureSocketsTransport_Disconnect() failed to close "
+                            "the connection with status code %d.",
+                            ( int ) xNetworkStatus ) );
+                xDemoStatus = pdFAIL;
+            }
+
+            xIsConnectionEstablished = pdFALSE;
+        }
+
+        /* Always stop the keep-alive timer for the next iteration. */
+        xTimerStatus = xTimerStop( xKeepAliveTimer, 0 );
+
+        if( xTimerStatus != pdPASS )
+        {
+            LogError( ( "xTimerStart() failed to stop KeepAliveTimer." ) );
+            xDemoStatus = pdFAIL;
+        }
```

Comment:
Can avoid the use of `xTimerStatus`
```suggestion
        xDemoStatus = xTimerStop( xKeepAliveTimer, 0 );

        if( xDemoStatus != pdPASS )
        {
            LogError( ( "xTimerStart() failed to stop KeepAliveTimer." ) );
        }
```

Reply:
Sure